mlflow/langchain/langchain_tracer.py

```diff
@@ -376,19 +377,28 @@ def on_tool_start(
         """Start span for a tool run."""
         if metadata:
             kwargs.update({"metadata": metadata})
+
+        # For function calling, input_str can be a stringified dictionary
+        # like "{'key': 'value'}". We try parsing it for better rendering,
+        # but conservatively fallback to original if it fails.
+        try:
+            inputs = ast.literal_eval(input_str)
+        except Exception:
+            inputs = input_str
+
         self._start_span(
             span_name=name or self._assign_span_name(serialized, "tool"),
             parent_run_id=parent_run_id,
             span_type=SpanType.TOOL,
             run_id=run_id,
-            inputs=input_str,
+            inputs=inputs,
```

Comment:
can we update the type hint on `inputs` of `_start_span`?

Reply:
Well the `inputs` argument passed to the handler is dictionary or `None`....  there seems to be some interesting context (and I suppose this is why we use `input_str`)

https://github.com/langchain-ai/langchain/blob/master/libs/core/langchain_core/tools/base.py#L636-L640

Reply:
So we don't need to update this line? We pass either `ast.literal_eval(input_str)` (dict) or `input_str` (str).

https://github.com/mlflow/mlflow/blob/cb578926f08f74ffcda7690dc8b61a61ce7c7462/mlflow/langchain/langchain_tracer.py#L82

Reply:
Ahh sorry I thought you meant the `inputs` of `on_tool_start`. Yea we should update the one for `start_span`.